### PR TITLE
Hotel service

### PR DESCRIPTION
**Pull request by moficodes**
Add hotel service
- get data from generated hotel data
- add filter for filtering by superchain, hotel name, hotel type and cost

Add lint test to travis

Add swagger docs for hotel service.

TODO: Add docker file and k8s configs.

### CODE REVIEW

**Comment by MaxShapiro32** [comment]
@moficodes Are the hotel images not local for v1?

File: src/hotel/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          required: true
+        - in: query
+          name: superchain
+          type: string
+          description: Filter by superchain name
+          required: false
+        - in: query
+          name: hotel
+          type: string
+          description: Filter by superchain name
```
Comment (MaxShapiro32) [inline src/hotel/swagger.yaml]: The filter descriptions should be changed

File: src/hotel/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        - Hotels
+      summary: Gets info data for Hotels
+      description: ''
+      produces:
+        - application/json
+      parameters:
+        - in: path
+          type: string
+          name:  topic
+          description: Topic to find info on superchains|hotels|hotel-info
```
Comment (MaxShapiro32) [inline src/hotel/swagger.yaml]: What is an example for what topic should be?

File: src/hotel/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+info:
+  description: >-
+    This is a microservice that handles hotel data for Bee Travel
+  version: 1.0.0
+  title: Bee Travel Hotel Data Service
+  contact:
+    email: moficodes@gmail.com
+  license:
+    name: Apache 2.0
+    url: 'https://github.com/bee-travels/bee-travels/blob/master/LICENSE'
```
Comment (MaxShapiro32) [inline src/hotel/swagger.yaml]: `bee-travels-node`

File: src/hotel/src/service/hotelService.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+var hotels = require(process.env.INIT_CWD +"/hotel-data.json");
+var hotelInfo = require(process.env.INIT_CWD +"/hotel-info.json");
+
+function getHotels(city, country, f) {
```
Comment (MaxShapiro32) [inline src/hotel/src/service/hotelService.js:4]: There are duplicate hotels for cities but with different prices. Is that intentional?

File: src/hotel/src/service/hotelService.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      return matchSuperchain && matchHotel && matchType && costHigherThan && costLowerThan;
+    });
+  }
+
+  return data;
+}
+
+
+
+function getHotelWithFilter() {
```
Comment (MaxShapiro32) [inline src/hotel/src/service/hotelService.js]: Is this going to be used later?

File: src/hotel/swagger.yaml (hunk 2 above)
Comment (moficodes) [reply src/hotel/swagger.yaml]: superchains | hotels | hotel-type

File: src/hotel/swagger.yaml (hunk 1 above)
Comment (moficodes) [reply src/hotel/swagger.yaml]: What should I change this too.

File: src/hotel/swagger.yaml (hunk 3 above)
Comment (moficodes) [reply src/hotel/swagger.yaml]: ahh. yeah. would update. 
in my defense: this url still works.

File: src/hotel/src/service/hotelService.js (hunk 4 above)
Comment (moficodes) [reply src/hotel/src/service/hotelService.js:4]: yes. think of hotels in primo locations.

File: src/hotel/src/service/hotelService.js (hunk 5 above)
Comment (moficodes) [reply src/hotel/src/service/hotelService.js]: i managed to get it working without using the extra filter. 😂 
will remove.

File: src/hotel/swagger.yaml (hunk 1 above)
Comment (moficodes) [reply src/hotel/swagger.yaml]: ahh copy paste error.

**Comment by MaxShapiro32** [comment]
@moficodes Should also add a README

**Review by moficodes — COMMENTED** [review]
all changes made

File: src/hotel/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Bee Travels Destination Service
```
Comment (MaxShapiro32) [inline src/hotel/README.md]: Destination -> Hotel

File: src/hotel/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+The hotel service is a microservice designed to provide information about various hotel locations for the Bee Travels travel application.
+
+# How to Run
+
+Follow these steps to setup and run the hotel service
+
+1. [Prerequisites](#1-prerequisites)
+2. [Clone the repo](#2-clone-the-repo)
+3. [Run the service](#4-run-the-service)
```
Comment (MaxShapiro32) [inline src/hotel/README.md]: 4 -> 3

File: src/hotel/swagger.yaml [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+    url: 'https://github.com/bee-travels/bee-travels-node/blob/master/LICENSE'
+host: 'localhost:9002'
+basePath: /api/v1/hotels
+tags:
+  - name: Hotels
+    description: Operations associated with hotel data
+schemes:
+  - http
+paths:
+  /hotels/{city}/{country}:
```
Comment (MaxShapiro32) [inline src/hotel/swagger.yaml]: The `basePath` on line 13 has a `/hotels` in it already so the `/hotels` isn't needed here
ex. `http://localhost:9002/api/v1/hotels/hotels/Los%20Angeles/United%20States`

**Review by MaxShapiro32 — APPROVED** [review]

**Pull request merged**